File: src/125. Palindromic sums/125.rb
require 'set'

class Integer
  def palindrome?
    self.to_s.reverse == self.to_s
  end
end

squares = [0]
1.upto(10 ** 4) {|i| squares << i * i}

result = Set.new
1.upto(squares.size - 1) do |i|
  sum = squares[i]
  (i + 1).upto(squares.size - 1) do |j|
    sum += squares[j]
    break if sum > 10 ** 8
    result.add(sum) if sum.palindrome?
  end
end
puts result.each.inject(:+)

File: src/125. Palindromic sums/125.py
def is_palindrome(n):
    return str(n)[::-1] == str(n)

squares = [0]
for i in range(1, 10 ** 4):
    squares.append(i * i)

result = set()
for i in range(1, len(squares)):
    n = squares[i]
    for j in range(i + 1, len(squares)):
        n += squares[j]
        if n > 10 ** 8:
            break
        if is_palindrome(n):
            result.add(n)

print(sum(result))
